core/controllers/editor_test.py

```diff
@@ -266,38 +275,50 @@ def _create_training_data(*arg):
                 exploration_dict['exploration']['states'][state_name][
                     'interaction']['id'], 'TextInput')
 
+            answer_groups = exp_services.get_exploration_by_id('15').states[
+                state_name].interaction.answer_groups
+            explicit_rule_spec_string = (
+                answer_groups[0].rule_specs[0].stringify_classified_rule())
+            fuzzy_rule_spec_string = (
+                answer_groups[1].rule_specs[0].stringify_classified_rule())
+
             # Input happy since there is an explicit rule checking for that.
-            self.submit_answer('15', state_name, 'happy')
+            event_services.AnswerSubmissionEventHandler.record(
+                exp_id, 1, state_name, explicit_rule_spec_string, 'happy')
 
             # Input text not at all similar to happy (default outcome).
-            self.submit_answer('15', state_name, 'sad')
+            event_services.AnswerSubmissionEventHandler.record(
+                exp_id, 1, state_name, exp_domain.DEFAULT_RULESPEC_STR, 'sad')
 
             # Input cheerful: this is current training data and falls under the
             # fuzzy rule.
-            self.submit_answer('15', state_name, 'cheerful')
+            event_services.AnswerSubmissionEventHandler.record(
+                exp_id, 1, state_name, fuzzy_rule_spec_string, 'cheerful')
 
             # Input joyful: this is not training data but will be classified
             # under the fuzzy rule.
-            self.submit_answer('15', state_name, 'joyful')
+            event_services.AnswerSubmissionEventHandler.record(
+                exp_id, 1, state_name, fuzzy_rule_spec_string, 'joyful')
 
             # Log in as an editor.
             self.login(self.EDITOR_EMAIL)
-            response = self.testapp.get('/create/15')
+            response = self.testapp.get('/create/%s' % exp_id)
             csrf_token = self.get_csrf_token_from_response(response)
-            url = str('/createhandler/training_data/15/%s' % state_name)
+            url = str(
+                '/createhandler/training_data/%s/%s' % (exp_id, state_name))
 
             exploration_dict = self.get_json(
-                '%s/15' % feconf.EXPLORATION_INIT_URL_PREFIX)
+                '%s/%s' % (feconf.EXPLORATION_INIT_URL_PREFIX, exp_id))
 
             # Only two of the four submitted answers should be unhandled.
             response_dict = self.get_json(url)
             self.assertEqual(
                 response_dict['unhandled_answers'],
-                _create_training_data('joyful', 'sad'))
+                _create_training_data('sad', 'joyful'))
```

Comment:
Yes, at initial glance this makes me slightly uncomfortable. The order for the response training data is very specific (answers are returned based on frequency). The tiebreaker is not clear to me, but I think it's deterministic and based on insertion order. Since insertion order didn't change, I'm slightly confused as well. @wxyxinyu when you address this, I can help you look into it more if you'd like. It might also not be an issue since this is largely based on a nearly deprecated answers storage system.